synapse/appservice/__init__.py

```diff
@@ -172,12 +172,24 @@ async def _matches_user_in_member_list(
         Returns:
             True if this service would like to know about this room.
         """
-        member_list = await store.get_users_in_room(
+        # We can use `get_local_users_in_room(...)` here because an application service
+        # can only be interested in local users of the server it's on (ignore any remote
+        # users that might match the user namespace regex).
+        #
+        # In the future, we can consider re-using
+        # `store.get_app_service_users_in_room` which is very similar to this
+        # function but has a slightly worse performance than this because we
+        # have an early escape-hatch if we find a single user that the
+        # appservice is interested in. The juice would be worth the squeeze if
+        # `store.get_app_service_users_in_room` was used in more places besides
+        # an experimental MSC. But for now we can avoid doing more work and
```

Comment:
[MSC3905](https://github.com/matrix-org/matrix-spec-proposals/pull/3905) is no longer experimental.

Reply:
This is talking about `store.get_app_service_users_in_room` usage which is only used by `_compute_msc3202_otk_counts_and_fallback_keys`

https://github.com/matrix-org/synapse/blob/d902181de98399d90c46c4e4e2cf631064757941/synapse/appservice/scheduler.py#L293-L330